nbgitpuller/pull.py

```diff
@@ -70,11 +70,60 @@ def __init__(self, git_url, branch_name, repo_dir, **kwargs):
         assert git_url and branch_name
 
         self.git_url = git_url
-        self.branch_name = branch_name
+
+        if branch_name == "None":
```

Comment:
```suggestion
        if branch_name is None:
```

Since `None` is a special object in Python, rather than a string.

Reply:
It isn't the NoneType type that gets pased back through the handlers. I drove myself in a couple circles trying figure out why "is None" wasn't working and realized the handlers were passing Strings. Maybe I am missing something here.

Reply:
I dug around some more. `SyncHandler` services the `api` endpoint, which is called by the JS in `static/index.js`. I see the following request being made by JS: `http://localhost:8888/git-pull/api?repo=https://github.com/yuvipanda/requirements&branch=None&targetpath=requirements`. See that branch is explicitly set to `None` - so it's something in the JS.

Reply:
The JS actually gets this from Python, in https://github.com/jupyterhub/nbgitpuller/blob/814682bb1a3452212212bed8079bf3f6a8045bce/nbgitpuller/static/index.js#L132.

Reply:
And here we put it in the HTML, in Python: https://github.com/jupyterhub/nbgitpuller/blob/814682bb1a3452212212bed8079bf3f6a8045bce/nbgitpuller/templates/status.html#L8.

You can see that `depth` is treated as optional, and only put in there if it is not None. However, branch is not considered as optional, and put in there all the time! So we need to add a similar conditional to branch, and only add it there if the branch argument is present. 

https://github.com/jupyterhub/nbgitpuller/blob/814682bb1a3452212212bed8079bf3f6a8045bce/nbgitpuller/static/index.js#L45 will also need to be amended to make `branch` operate in a similar fashion to `depth`, and https://github.com/jupyterhub/nbgitpuller/blob/814682bb1a3452212212bed8079bf3f6a8045bce/nbgitpuller/handlers.py#L55 as well. There might be a couple other places that assume `branch` is always present, we'll have to track those down too.